clang/lib/Sema/SemaDecl.cpp

```diff
@@ -6107,6 +6109,44 @@ static bool isFromSystemHeader(SourceManager &SM, const Decl *D) {
          SM.isInSystemMacro(D->getLocation());
 }
 
+constexpr unsigned countCPlusPlusKeywords() {
+  unsigned Ret = 0;
+#define MODULES_KEYWORD(NAME)
+#define KEYWORD(NAME, FLAGS) ++Ret;
+#define CXX_KEYWORD_OPERATOR(NAME, TOK) ++Ret;
+#include "clang/Basic/TokenKinds.def"
+  return Ret;
+}
+
+static bool isKeywordInCPlusPlus(const Sema &S, const IdentifierInfo *II) {
+  if (!II)
+    return false;
+
+  // Build a static map of identifiers for all of the keywords in C++ that are
+  // not keywords in C. This allows us to do pointer comparisons instead of
+  // string comparisons when deciding whether the given identifier is a keyword
+  // or not. Note, this treats all keywords as being enabled, regardless of the
+  // setting of other language options. It intentionally disables the modules
+  // keywords because those are conditional keywords, so may be safe to use.
+  static auto Keywords = [&S] {
+    std::array<uintptr_t, countCPlusPlusKeywords()> Ret;
+    unsigned Idx = 0;
+#define MODULES_KEYWORD(NAME)
+#define KEYWORD(NAME, FLAGS)                                                   \
+  Ret[Idx++] = reinterpret_cast<uint64_t>(                                     \
+      &S.getPreprocessor().getIdentifierTable().get(#NAME));
```

Comment:
This will break for at least all users of Clang as a library -- we can't store pointers to `IdentifierInfo`s in a static local table, because we can have multiple preprocessors (and hence identifier tables and `IdentifierInfo`s for a single identifier) in a process.